torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3960,6 +3960,10 @@ def track_symfloat(source, val):
             else:
                 sources_tensors_constraints = [(source, t, context.constraint_sizes)]
 
+                from torch.nested._internal.nested_tensor import _tensor_symint_registry
+                if t in _tensor_symint_registry:
+                    track_symint(torch._dynamo.source.NestedIntSource(source), _tensor_symint_registry[t])
```

Comment:
Part of the reason I'm doing it this way vs. unbacked symints is that unbacked symint approach does not work (it worked at one point). I'm still working through those issues.

It may not necessary for this PR actually, but two cases where we eventually need to have a non-unbacked symint is:

(1) In this case, offset1 and offset2 are not the same tensor object, so we cannot rely on dynamo graph arg deduping guards to trigger. 
```
trust_me_assert_equal(offset1, offset2)

def fn(values1, offset1, values2, offset2):
    nt1 = make_nt(values1, offset1)
    nt2 = make_nt(values2, offset2)
    ...
```

(2) Factory functions